src/reconcile/rec_write.c

```diff
@@ -553,14 +567,18 @@ __rec_write_check_complete(WT_SESSION_IMPL *session, WT_RECONCILE *r)
 		return (EBUSY);
 
 	/*
-	 * If we are doing eviction and restoring updates, there is only one
-	 * block and all update were skipped, no progress has been made and
-	 * there is no point swapping the new page into place.
+	 * If we are doing update/restore based eviction, confirm part of the
+	 * page is being discarded, or at least 10% of the updates won't have
```

Comment:
10% appears a bit arbitrary, I'm a little concerned that a workload will fail to make progress because of this change. OTOH I don't like wasting CPU on update-restore that isn't useful, so I see a definite benefit.

I'm inclined to merge, but watch test/format stress testing, and take a closer look at this check if we start seeing timeouts.